lib/array_intersection.rb

```diff
@@ -1,3 +1,11 @@
 def intersection(list1, list2)
-  
+  result = []
+  hash = {}
```

Comment:
Consider using a variable name that is more descriptive of what the hash is being used for instead of using the name of the data structure. Some possibilities could be `list1_values`, `list1_values_hashed`.